File: Python3/프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    answer = []
    for i in queries:
        m = []
        for j in range(i[0],i[1]+1):
            if(arr[j]>i[2]):
                m.append(arr[j])
        if (len(m)==0):
            answer.append(-1)
        else:
            answer.append(min(m))
    return answer

File: 기음#/프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.cs
using System;

public class Solution {
    public int[] solution(int[] arr, int[,] queries) {
        int[] answer = new int[queries.GetLength(0)];
        for (int i=0;i<queries.GetLength(0);i++){
            int  min = 1000001;
            for (int j=queries[i,0];j<=queries[i,1];j++){
                if (arr[j]>queries[i,2]){
                    min = arr[j]<min?arr[j]:min;
                }
            }
            answer[i] = min==1000001?-1:min;
        }
        return answer;
    }
}